crates/sncast/src/helpers/interactive.rs

```diff
@@ -0,0 +1,266 @@
+use anyhow::{Context, Result};
+use camino::Utf8PathBuf;
+use configuration::search_config_upwards_relative_to;
+use dialoguer::theme::ColorfulTheme;
+use dialoguer::Select;
+use std::fmt::Display;
+use std::{env, fs};
+use toml_edit::{DocumentMut, Item, Table, Value};
+
+enum PromptSelection {
+    GlobalDefault(Utf8PathBuf),
+    LocalDefault(Utf8PathBuf),
+    No,
+}
+
+impl Display for PromptSelection {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PromptSelection::LocalDefault(path) => {
+                write!(f, "Yes, local default ({})", to_tilde_path(path))
+            }
+            PromptSelection::GlobalDefault(path) => {
+                write!(f, "Yes, global default ({})", to_tilde_path(path))
+            }
+            PromptSelection::No => write!(f, "No"),
+        }
+    }
+}
+
+pub fn prompt_to_add_account_as_default(account: &str) -> Result<()> {
+    let mut options = Vec::new();
+
+    if let Ok(global_path) = get_global_config_path() {
+        options.push(PromptSelection::GlobalDefault(global_path));
+    }
+
+    if let Some(local_path) = env::current_dir()
+        .ok()
+        .and_then(|current_path| Utf8PathBuf::from_path_buf(current_path.clone()).ok())
+        .and_then(|current_path_utf8| search_config_upwards_relative_to(&current_path_utf8).ok())
+    {
+        options.push(PromptSelection::LocalDefault(local_path));
+    }
+
+    options.push(PromptSelection::No);
+
+    let selection = Select::with_theme(&ColorfulTheme::default())
+        .with_prompt("Do you want to make this account default?")
+        .items(&options)
+        .default(0)
+        .interact()
+        .context("Failed to display selection dialog")?;
+
+    match &options[selection] {
+        PromptSelection::GlobalDefault(path) | PromptSelection::LocalDefault(path) => {
+            edit_config(path, "default", "account", account)?;
+        }
+        PromptSelection::No => {}
+    }
+
+    Ok(())
+}
+
+fn edit_config(config_path: &Utf8PathBuf, profile: &str, key: &str, value: &str) -> Result<()> {
+    let file_content = fs::read_to_string(config_path).context("Failed to read config file")?;
+
+    let mut toml_doc = file_content
+        .parse::<DocumentMut>()
+        .context("Failed to parse TOML")?;
+    update_config(&mut toml_doc, profile, key, value);
+
+    fs::write(config_path, toml_doc.to_string()).context("Failed to write to config file")?;
+
+    Ok(())
+}
+
+fn update_config(toml_doc: &mut DocumentMut, profile: &str, key: &str, value: &str) {
+    if !toml_doc.contains_key("sncast") {
+        toml_doc["sncast"] = Item::Table(Table::new());
+    }
+
+    let sncast_table = toml_doc
+        .get_mut("sncast")
+        .and_then(|item| item.as_table_mut())
+        .expect("Failed to create or access 'sncast' table");
+
+    if !sncast_table.contains_key(profile) {
+        sncast_table[profile] = Item::Table(Table::new());
+    }
```

Comment:
Wouldn't that solve the problem with empty `[sncast]` table being created?

```suggestion
    let sncast_table = toml_doc
        .get_mut(format!("sncast.{profile}"))
        .and_then(|item| item.as_table_mut())
        .expect("Failed to create or access 'sncast' table");
```

Reply:
Nope